src/coreclr/jit/compiler.hpp

```diff
@@ -1081,10 +1081,9 @@ inline GenTreeIndir* Compiler::gtNewIndexIndir(GenTreeIndexAddr* indexAddr)
     else
```

Comment:
Is the check above missing out on creating an `Indir` node for `TYP_SIMD`?

Reply:
Yes. Another case of unnecessarily non-canonical representation that I plan to fix.